### PR TITLE
docs: deprecate random_string() basic,md5,sha1

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Supersedes #7327
See https://forum.codeigniter.com/showthread.php?tid=87026

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: user_guide_src/source/changelogs/v4.3.3.rst [hunk 1]
```diff
@@ -28,6 +28,9 @@ Changes
 Deprecations
 ************
 
+- **Text Helper:** :php:func:`random_string()`'s type ``basic``, ``md5``, and
```
Comment (datamweb) [inline user_guide_src/source/changelogs/v4.3.3.rst:31]:
```suggestion
- **Text Helper:** :php:func:`random_string()`'s type ``basic``, ``md5`` and
```

Comment (MGatner) [reply user_guide_src/source/changelogs/v4.3.3.rst:31]: I am in favor of keeping the [Oxford Comma](https://www.grammarbook.com/blog/commas/oxford-comma/) 😊

Comment (datamweb) [reply user_guide_src/source/changelogs/v4.3.3.rst:31]: As always something new to learn, great link. 
Thanks

**Comment by kenjis** [comment]
Need to update the changelog version.

**Comment by kenjis** [comment]
Rebased.

**Review by datamweb — APPROVED** [review]

**Pull request merged**